language/tools/cost-synthesis/src/stack_generator.rs

```diff
@@ -831,18 +840,23 @@ where
         })
     }
 
+    pub fn execute_code_snippet(
```

Comment:
@tzakian it is a bit odd to have this here but again it's the most natural place given ownership of data. I think it's fine but please keep it in mind